mysql-test/suite/galera/t/galera_bf_abort_wsrep_on.inc

```diff
@@ -0,0 +1,92 @@
+--source include/have_debug_sync.inc
+--source include/galera_cluster.inc
+--source include/count_sessions.inc
+
+#
+# Test a transaction within session with wsrep_on set to on/off being aborted
+# by TOI transaction
+#
+
+# common setup
+--connect node_1a, 127.0.0.1, root, , test, $NODE_MYPORT_1
+--connection node_1
+CREATE TABLE t1(id INT PRIMARY KEY AUTO_INCREMENT, k INT);
+INSERT INTO t1(k) VALUES (1),(2),(3),(101),(102),(103);
+--eval SET SESSION wsrep_on = $wsrep_on_value;
+
+#
+# Case 1: Abort by local TOI when query in IDLE state
```

Comment:
In this we have coverage for `IDLE` and `EXEC`. Can we also add one mode scenario for `QUERY_COMMITTING` state? I think we can make use of the `wsrep_before_certification` in `sql/wsrep_hton.cc` debug_sync_point.

Reply:
When wsrep_on=0, the query is not replicated, so it is not entering QUERY_COMMITTING state. However, your comment gave me the hint for improving MTR test by introducing 2 more cases and handling 1 corner case :)